src/main/java/mate/academy/service/UserService.java

```diff
@@ -3,18 +3,27 @@
 import mate.academy.model.User;
 
 public class UserService {
-    private static final User[] users = new User[] {
+    private static final User[] users = new User[]{
             new User("bob@i.ua", "1234"),
             new User("alice@i.ua", "1234")
     };
 
     /**
      * Find user by email. All users are stored in <code>private static final User[] users</code>
+     *
      * @param email - the input parameter
      * @return - user if his email is equal to passed email.
      * Return <code>null</code> if there is no suitable user
      */
-    public User findByEmail(String email) {
-        return null;
+    public static User findByEmail(String email) {
+        User foundUser = null;
+        for (User user: users) {
+            String userEmail = user.getEmail();
+            if (userEmail == email) {
+                foundUser = user;
+                break;
+            }
+        }
+        return foundUser;
```

Comment:
returning early is a better pattern, more accepted in the [community](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement)

Reply:
++